packages/govuk-frontend/src/govuk/govuk-frontend-component.mjs

```diff
@@ -29,4 +29,37 @@ export class GOVUKFrontendComponent {
       throw new SupportError()
     }
   }
+
+  /**
+   *
+   * @protected
+   * @param {Schema} schema
+   * @param {object} config
+   * @returns {boolean} Whether the config is valid
+   */
+  validateConfig(schema, config) {
+    if (schema.anyOf) {
+      // Check if at least one of the conditions in anyOf is met
+      for (const condition of schema.anyOf) {
+        const requiredProps = condition.required || []
+        if (requiredProps.every((prop) => prop in config)) {
+          return true
+        }
+      }
+      return false
+    }
+    return true
+  }
 }
+
+/**
+ * @typedef {object} Schema
+ * @property {Array<SchemaItem>} [anyOf] - An array of conditions
```

Comment:
I'm sure ESLint used to ask for shorthand at some point?

Would be this if you prefer it:

```suggestion
 * @property {SchemaItem[]} [anyOf] - An array of conditions
```

Reply:
Ugh, I do prefer it, I need to change my snippets. But eslint not complaining for me.